scripts/generate-changelog.js

```diff
@@ -0,0 +1,66 @@
+'use strict'
+
+const replace = require('replace-in-file')
+const util = require('util')
+const exec = util.promisify(require('child_process').exec)
+const spawn = util.promisify(require('child_process').spawn)
+
+const name = require('emoji-name-map')
+
+const lernaConfig = require('../lerna.json')
+
+const CHANGELOG_FILE = 'CHANGELOG.md'
+
+async function main() {
+  const lastTagHash = await executeCommand('git rev-list --tags --max-count=1')
+  const lastTagName = await executeCommand(`git describe --tags ${lastTagHash}`)
+
+  const oldTag = lastTagName.replace(/\n$/, '')
+  const commits = await executeCommand(`git log ${oldTag}..HEAD --pretty=format:"- %s"`)
+
+  const changesWithEmojis = await emojiNameToUnicode(`# Changelog\n\n## v${lernaConfig.version}\n\n${commits}`)
+
+  const changesWhihPullRequestLinks = changesWithEmojis.replace(
+    /\(#(\d+)\)/gm,
+    (match, id) => `([#${id}](https://github.com/DataDog/browser-sdk/pull/${id}))`
+  )
+
+  await replace({
+    files: CHANGELOG_FILE,
+    from: new RegExp('# Changelog', 'g'),
+    to: changesWhihPullRequestLinks,
+  })
+
+  const openEditorCmd = await spawn(process.env.EDITOR, [CHANGELOG_FILE], { stdio: 'inherit', detached: true })
+  if (openEditorCmd.stderr) {
+    throw openEditorCmd.stderr
+  }
+}
+
+async function executeCommand(command) {
+  const commandResult = await exec(command)
+  if (commandResult.stderr) {
+    throw commandResult.stderr
+  }
+  return commandResult.stdout
+}
+
+async function emojiNameToUnicode(changes) {
+  const emojiNameRegex = new RegExp(/:[^:\s]*(?:::[^:\s]*)*:/, 'gm')
+
+  let matches
+  while ((matches = emojiNameRegex.exec(changes))) {
+    if (!!matches) {
+      await matches.map((match) => {
+        changes = changes.replace(match, name.get(match) || match)
+      })
+    }
```

Comment:
FMU, it should loop forever if there is a missing emoji in the map. Am I missing something?

Reply:
It will not loop forever, because `emojiNameRegex` is stateful and will start its execution after its `lastIndex` property. But still, this loop should be changed:
* No need to be async
* No need to check for `!!matches` since it wont execute the loop body if `matches` were falsy
* No need to use `.map` since we don't return anything (you could use `matches[0]` since there is only one element in matches)
* Mutating a string while iterating over it using a regex will probably lead to bugs

Reply:
Why not just using
```js
return changes.replace(emojiNameRegex, (emoji) => name.get(match) || match)
```

Reply:
> It will not loop forever, because emojiNameRegex is stateful and will start its execution after its lastIndex property. 

What do you mean by lastIndex?
My point was not about the regex but the fact that we are iterating on emoji matches while we can fail to replace them all (`replace(match, name.get(match) || match)`)

Reply:
A loop like this will always finish at some point, because the regex object keeps an ever-increasing index and will return null at some point:

```
const s = "aa";
const re = /a/g;

re.lastIndex // === 0
re.exec(s) // ["a"]
re.lastIndex // === 1
re.exec(s) // ["a"]
re.lastIndex // === 2
re.exec(s) // null -> this will break the loop
re.lastIndex // === 0
```

Reply:
I was not aware of this api, good to know 👍